.github/workflows/nightly-e2e-tests.yml

```diff
@@ -0,0 +1,39 @@
+name: nightly-e2e-tests
+
+on:
+  schedule:
+    - cron:  '* 23 * * *'
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [macos-latest, windows-latest]
```

Comment:
ubuntu is triggered by the commits fair but I would also include it in the nightly test just in case we do not work on it for a longer time we potentially miss an error.